src/App.tsx

```diff
@@ -1,43 +1,47 @@
-import { useEffect, useState } from 'react';
+import { useEffect } from 'react';
 import {
 	BrowserRouter as Router,
 	Routes,
 	Route,
 	Navigate,
 } from 'react-router-dom';
+import { legacy_createStore as createStore } from 'redux';
+import { useDispatch, useSelector } from 'react-redux';
+import { composeWithDevTools } from 'redux-devtools-extension';
 
+import rootReducer from './store';
 import Header from './components/Header';
 import Courses from './components/Courses';
 import CreateCourse from './components/CreateCourse';
 import CourseInfo from './components/CourseInfo';
 import Registration from './components/Registration';
 import Login from './components/Login/Login';
-import { mockedAuthorsList, mockedCoursesList } from './constants';
-import type { Course, Author } from './helpers/interfaces';
+import { getCourses } from './store/courses/actionCreators';
+import { getAuthors } from './store/authors/actionCreators';
+import { selectUser } from './helpers';
+import type { User } from './helpers';
 
 import './App.css';
 
+const store = createStore(rootReducer, composeWithDevTools());
+type RootState = ReturnType<typeof store.getState>;
+
 const App: React.FC = () => {
-	const [courseList, setCourseList] = useState<Course[]>(mockedCoursesList);
-	const [authorList, setAuthorList] = useState<Author[]>(mockedAuthorsList);
-	const [userName, setUserName] = useState<string | null>('');
+	const dispatch = useDispatch();
+	const userName: User = useSelector(selectUser);
```

Comment:
I think type inferring can be applied here if you configured your selector correctly